app/controllers/proposals_controller.rb

```diff
@@ -2,6 +2,7 @@ class ProposalsController < ApplicationController
   include TokenAuth
 
   skip_before_action :authenticate_user!, only: [:approve]
+  before_action :check_disabled_client
```

Comment:
it's a trivial operation in terms of cpu cycles, and it future-proofs the feature for when the inevitable additional methods are added that do not use `authenticate_user!` and someone has to remember to add them to a whitelist of actions to check.